bento/src/main/java/com/yelp/android/bento/components/ListComponent.java

```diff
@@ -85,18 +85,31 @@ public void appendData(@NonNull List<T> data) {
         notifyItemRangeInserted(oldSize, sizeChange);
     }
 
+    /**
+     * Inserts the specified data at the specified index in the component data.
+     *
+     * @param index the index to insert the given data
+     * @param data the data to be inserted
+     */
+    public void insertData(int index, @NonNull T data) {
+        mData.add(index, data);
+        notifyItemRangeInserted(index, index + 1);
+    }
+
     /**
      * Removes the provided data items from the list.
      *
      * @param data The data item to remove from the list.
+     * @return an int representing the index that the removed data occupied
```

Comment:
nit: remove "an int representing". It's redundant due to the function signature.
Also, add something about returning -1 when nothing is removed.